File: app/src/main/java/org/deiverbum/app/model/Preces.java
package org.deiverbum.app.model;

import static org.deiverbum.app.utils.Constants.BRS;
import static org.deiverbum.app.utils.Utils.LS2;

import android.text.SpannableStringBuilder;
import android.text.Spanned;

import org.deiverbum.app.utils.Utils;

public class Preces {
    public String intro;
    public String texto;

    public String getIntro() {
        return intro;
    }

    public void setIntro(String intro) {
        this.intro = intro;
    }

    public SpannableStringBuilder getAll() {

        SpannableStringBuilder sb=new SpannableStringBuilder();
        sb.append(getHeader());
        sb.append(LS2);
        String[] introArray = intro.split("\\|");
        if (introArray.length == 3) {
            sb.append(introArray[0]);
            sb.append(LS2);
            sb.append(Utils.fromHtml(String.format("<i>%s</i>",introArray[1])));
            sb.append(LS2);
            sb.append(getTexto());
            sb.append(LS2);
            sb.append(introArray[2]);
        } else {
            sb.append(getIntro());
            sb.append(LS2);
            sb.append(getTexto());
        }
        return sb;

    }

    public Spanned getTexto() {
        return Utils.fromHtml(Utils.getFormato(texto));
    }


    public void setTexto(String texto) {
        this.texto = texto;
    }

    public SpannableStringBuilder getHeader() {

        return Utils.formatTitle("PRECES");
    }

    public String getHeaderForRead() {
        return "PRECES.";
    }


    /**
     * @deprecated desde la versión 2022.01 - Reemplazar por {@link Preces#getAll()}
     * @return
     */
    @Deprecated
    public String getPreces() {
        return "";
    }

    public SpannableStringBuilder getAllForRead() {

        SpannableStringBuilder sb=new SpannableStringBuilder();
        sb.append(getHeaderForRead());
        sb.append(LS2);
        String[] introArray = intro.split("\\|");
        if (introArray.length == 3) {
            sb.append(introArray[0]);
            sb.append(introArray[1]);
            sb.append(getTexto());
            sb.append(introArray[2]);
        } else {
            sb.append(getIntro());
            sb.append(getTexto());
        }
        return sb;

    }
}

File: app/src/main/java/org/deiverbum/app/core/model/data/ritualis/Preces.kt
package org.deiverbum.app.core.model.data.ritualis

import android.text.SpannableStringBuilder
import org.deiverbum.app.util.Constants
import org.deiverbum.app.util.Constants.CHAR_R
import org.deiverbum.app.util.Constants.PRECES_R
import org.deiverbum.app.util.Utils
import org.deiverbum.app.util.Utils.LS2

data class Preces(
    override var type: String = "preces",
    var intro: String? = null,
    var responsum: String,
    var preces: List<String>

) : Content(type) {
    override fun forView(): SpannableStringBuilder {
        val ssb = SpannableStringBuilder()
        if (intro != null) {
            ssb.append(Utils.fromHtml(intro!!))
            ssb.append(": ${responsum}")
            ssb.append(LS2)
        }
        for (t in preces) {
            ssb.append(Utils.toRed(PRECES_R))
            ssb.append(t)
            ssb.append(LS2)
            ssb.append(Utils.toRed(CHAR_R))
            ssb.append(responsum)

            //ssb.append(" ")
            //ssb.append(Utils.toRed(pericopa))
            ssb.append(Constants.LS2)
        }
        return ssb


    }
}